src/actor.js

```diff
@@ -524,13 +496,26 @@ export const callTask = async (taskId, input, options = {}) => {
     if (waitSecs <= 0) return run; // In this case there is nothing more to do.
 
     // Wait for run to finish.
-    const updatedRun = await waitForRunToFinish({
-        actId: run.actId,
-        runId: run.id,
-        token,
-        waitSecs,
-        taskId,
-    });
+    let updatedRun;
+    try {
+        updatedRun = await waitForRunToFinish({
+            actorId: run.actId,
+            runId: run.id,
+            token,
+            waitSecs,
+        });
+    } catch (err) {
+        if (err.message.startsWith('Waiting for run to finish')) {
```

Comment:
Somewhere I read that parsing exception error messages is really dirty and prone to breaks :) Maye you can just test if that's ApifyCallError instance...

Reply:
But it's not an `ApifyCallError` instance, because the error is thrown from `waitForRunToFinish` which can now be called separately from `Apify.call` 😅 

Let's wait for the error redesign. I didn't want to make a `WaitForRunToFinishError` that someone would potentially depend on, before we decide on how to actually handle the errors. #646

Reply:
Cool, another way would be just to define some Symbol, set it to the error and then check its presence in catch